hubble: broken build

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *hubble*

### Build log

https://hydra.nixos.org/build/180878959/nixlog/2

```
=== RUN   Test_getFlowsRequestWithInvalidRawFilters
    flows_test.go:102: 
                Error Trace:    flows_test.go:102
                Error:          Not equal: 
                                expected: "invalid --allowlist flag: failed to decode '{\"invalid\":[\"filters\"]}': proto:\u00a0(line 1:2): unknown field \"invalid\""
                                actual  : "invalid --allowlist flag: failed to decode '{\"invalid\":[\"filters\"]}': proto: (line 1:2): unknown field \"invalid\""
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -invalid --allowlist flag: failed to decode '{"invalid":["filters"]}': proto: (line 1:2): unknown field "invalid"
                                +invalid --allowlist flag: failed to decode '{"invalid":["filters"]}': proto: (line 1:2): unknown field "invalid"
                Test:           Test_getFlowsRequestWithInvalidRawFilters
    flows_test.go:104: 
                Error Trace:    flows_test.go:104
                Error:          Not equal: 
                                expected: "invalid --denylist flag: failed to decode '{\"invalid\":[\"filters\"]}': proto:\u00a0(line 1:2): unknown field \"invalid\""
                                actual  : "invalid --denylist flag: failed to decode '{\"invalid\":[\"filters\"]}': proto: (line 1:2): unknown field \"invalid\""
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -invalid --denylist flag: failed to decode '{"invalid":["filters"]}': proto: (line 1:2): unknown field "invalid"
                                +invalid --denylist flag: failed to decode '{"invalid":["filters"]}': proto: (line 1:2): unknown field "invalid"
                Test:           Test_getFlowsRequestWithInvalidRawFilters
--- FAIL: Test_getFlowsRequestWithInvalidRawFilters (0.00s)
=== RUN   Test_getFlowFiltersYAML
--- PASS: Test_getFlowFiltersYAML (0.00s)
=== RUN   Test_getFlowsBasic
--- PASS: Test_getFlowsBasic (0.00s)
=== RUN   Test_getFlowsTimeRange
--- PASS: Test_getFlowsTimeRange (0.00s)
=== RUN   Test_getFlowsFilter
--- PASS: Test_getFlowsFilter (0.00s)
FAIL
FAIL    github.com/cilium/hubble/cmd/observe    0.018s
FAIL
```

### Additional context

Maybe related to https://github.com/cilium/hubble/pull/656 and https://github.com/cilium/hubble/pull/655

I've tried to build with Go 1.18 inside nix expression and get the same error, a nix-shell with cmake and go_1_18 can build it OK.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@humancalico 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.14, NixOS, 22.11 (Raccoon), 22.11.20220615.b195759`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.1`
 - channels(root): `""`
 - channels(bryan): `""`
 - nixpkgs: `/etc/nix/channels/nixpkgs`
```

--- a/pkgs/applications/networking/cluster/hubble/default.nix
+++ b/pkgs/applications/networking/cluster/hubble/default.nix
@@ -1,8 +1,8 @@
-{ stdenv, lib, buildGoModule, fetchFromGitHub }:
+{ stdenv, lib, buildGoModule, fetchFromGitHub, installShellFiles }:
 
 buildGoModule rec {
   pname = "hubble";
-  version = "0.9.0";
+  version = "0.10.0";
 
   src = fetchFromGitHub {
     owner = "cilium";
@@ -13,11 +13,36 @@ buildGoModule rec {
 
   vendorSha256 = null;
 
+  ldflags = [
+    "-s" "-w"
+    "-X github.com/cilium/hubble/pkg.GitBranch=none"
+    "-X github.com/cilium/hubble/pkg.GitHash=none"
+    "-X github.com/cilium/hubble/pkg.Version=${version}"
+  ];
+
+  # Test fails at Test_getFlowsRequestWithInvalidRawFilters in github.com/cilium/hubble/cmd/observe
+  # https://github.com/NixOS/nixpkgs/issues/178976
+  # https://github.com/cilium/hubble/pull/656
+  # https://github.com/cilium/hubble/pull/655
+  doCheck = false;
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/hubble version | grep ${version} > /dev/null
+  '';
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    installShellCompletion --cmd hubble \
+      --bash <($out/bin/hubble completion bash) \
+      --fish <($out/bin/hubble completion fish) \
+      --zsh <($out/bin/hubble completion zsh)
+  '';
+
   meta = with lib; {
     broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;
     description = "Network, Service & Security Observability for Kubernetes using eBPF";
     license = licenses.asl20;
     homepage = "https://github.com/cilium/hubble/";
-    maintainers = with maintainers; [ humancalico ];
+    maintainers = with maintainers; [ humancalico bryanasdev000 ];
   };
 }

